plutus-scb/src/Cardano/Metadata/Client.hs

```diff
@@ -1,40 +1,59 @@
+import           Cardano.Metadata.Types    (JSONEncoding (ExternalEncoding),
+                                            MetadataEffect (GetProperties, GetProperty),
+                                            MetadataError (MetadataClientError))
 import           Control.Monad.Freer       (Eff, LastMember, Member, type (~>), interpret, sendM)
 import           Control.Monad.Freer.Error (Error, throwError)
 import           Control.Monad.IO.Class    (MonadIO, liftIO)
+import           Data.Coerce               (coerce)
 import           Data.Proxy                (Proxy (Proxy))
-import           Servant.Client            (ClientEnv, ClientError, ClientM, client, runClientM)
+import           Network.HTTP.Types        (status404)
+import           Servant.Client            (ClientEnv, ClientError (FailureResponse), ClientM, client,
+                                            responseStatusCode, runClientM)
 import           Servant.Extra             (left, right)
 
-getProperties :: Subject -> ClientM [Property]
-getProperty :: Subject -> PropertyKey -> ClientM Property
-(getProperties, getProperty) = (_getProperties, _getProperty)
-  where
-    _getProperties = left . api
-    _getProperty = right . api
-    api = client (Proxy @API)
-
 handleMetadataClient ::
        forall m effs.
-       (LastMember m effs, MonadIO m, Member (Error ClientError) effs)
+       ( LastMember m effs
+       , MonadIO m
+       , Member (Error MetadataError) effs
+       )
     => ClientEnv
     -> Eff (MetadataEffect ': effs) ~> Eff effs
 handleMetadataClient clientEnv =
-    let runClient :: forall a. ClientM a -> Eff effs a
-        runClient a =
-            (sendM $ liftIO $ runClientM a clientEnv) >>= either throwError pure
+    let (getProperties, getProperty) = (_getProperties, _getProperty)
+          where
+            _getProperties = left . api
+            _getProperty = right . api
+            -- This can be flipped between AesonEncoding and
+            -- ExternalEncoding depending on which metadata server
+            -- we're using.
+            api = client (Proxy @(API 'ExternalEncoding))
+
+        runClient :: ClientM (Maybe a) -> Eff effs (Maybe a)
+        runClient action = do
+              result <- sendM $ liftIO $ runClientM action clientEnv
+              case result of
+                Right value -> pure value
+                (Left (FailureResponse _ response)) | responseStatusCode response == status404 -> pure Nothing
+                (Left err) -> throwError $ MetadataClientError err
      in interpret $ \case
-            GetProperties subject -> runClient $ getProperties subject
-            GetProperty subject propertyKey ->
-                runClient $ getProperty subject propertyKey
+          GetProperties subject ->
+              coerce <$> runClient (getProperties subject)
+          GetProperty subject propertyKey ->
+              coerce <$> runClient (getProperty subject propertyKey)
```

Comment:
what's `coerce` doing here? maybe a comment or a type annotation?

Reply:
The docs I've added clarify this too.